addons/cert.py

```diff
@@ -380,16 +380,16 @@ def api01(data):
         if scope.type!='Struct':
             continue
         token = scope.bodyStart
-        arrayFound=False
+        arrayFound = None
         # loop through the complete struct
         while token != scope.bodyEnd:
             if token.isName and token.variable:
-                if token.variable.isArray:
-                    arrayFound=True
+                if token.variable.isArray and ('char' in token.variable.typeStartToken.str):
+                    arrayFound = token
                 elif arrayFound and not token.variable.isArray and not token.variable.isConst:
-                    reportError(token, 'style', 'Avoid laying out strings in memory directly before sensitive data', 'API01-C')
+                    reportError(arrayFound, 'style', 'Avoid laying out strings in memory directly before sensitive data', 'API01-C')
```

Comment:
This seems to break suppressions, but I'm not sure how/why. 

The intent of course was to point at the cstring for which this error is being raised. But in my own list of suppressions (a `*.cppcheck` project) I have to use the line number + 1, so still seems to be pointing at the next line, the "sensitive data" part. 

Happy to revert this part. I was just trying to help improve report clarity.